src/Template/Element/advanced_search_form.ctp

```diff
@@ -357,6 +357,10 @@ echo $this->Form->create('AdvancedSearch', [
                    href="http://en.wiki.tatoeba.org/articles/show/text-search">
             <?= __('More search options') ?>
         </md-button>
+
+        <md-button type="submit" class="md-primary" formaction="/eng/sentences/advanced_search">
+            <?= __('Permalink to search') ?>
```

Comment:
I don’t think this button should appear on the right pane of the search results.

In terms of UI, I’m not sure a button fits the use case after all. This button forces the user to reload the page (by clicking it) in order to access the actual link. I’d personally rather have a link on which I can either click, or right-click → copy URL.

I wonder what would be the best wording. It refers to the *search criteria*, not the *search results*, so "search" alone may be too ambiguous. How about "Permalink to this form"? Maybe we can go for something totally different like "Bookmark this search"; but the action of the button is not exactly bookmarking. @alanfgh any suggestions?

Reply:
How about just "create link"? It's appearing right next to two search buttons so I think it'll be clear that it's referring to the search.

Reply:
If I’d see a "Create link" button, I’d ask to myself: "what link?" I understand it’s related to search, but it’s still rather unclear. From there, the result of clicking on this "Create link" button is that I am brought to the same page. Only if I am curious enough to pay attention to the URL (which would probably be hidden on mobile), I may understand that I can re-use that URL to open the form with pre-filled values.

I think we may be able to make the whole thing easier to understand by showing more clearly what happens *after* clicking on the button. For example, you could display a flash message (after the page reloads) that says something among the lines "You can bookmark this page or copy its URL to load the search form with pre-filled values."

Reply:
Makes sense. How does this sound:

Label the button "create link" (or something else concise and somewhat vague)

When the button is clicked:
Use a little Javascript to copy the link to the clipboard without reloading, then display a small popup above the button that reads "Bookmark or share this link to load the search form with the values you've already filled in"

Reply:
Thanks for your feedback :smile: In your suggestion, the UI mentions a link, but that link is not visible anywhere (because it’s directly copied to the clipboard), so one might still ask "What link?"

Compare for example with how Youtube does it. They have a "share" button that opens a dialog containing a text field containing a link, with a "copy" button next to it.

Looking back to #1832, having the link automatically copied to the clipboard makes it easier to "share the search to a friend", but harder to "do the same research again" or "do a similar research" because the user will have to paste the link around instead of just clicking. However, I think only the latter was the main point of the ticket. I can see a lot more use cases for a pre-filled search form one can immediately re-use than for pasting the link somewhere or to somebody. I might be wrong/biased, but I can imagine how one would want do a lot of repeated searches, while I cannot really see why somebody would want to do a lot of repeated pasting.

The more I think about it, the more I think we shouldn’t mess up with the clipboard. I think a "Create a search template" button might be the best way to go because it can easily extend to "Create a search template that I can share" into one’s mind, whereas "Bookmark/share" won’t easily extend to "Bookmark/share a search template".

Reply:
I agree, redoing a similar/identical seems like a far more common use case, which a "copy link to clipboard" button wouldn't facilitate very well. Keeping things as is does make redoing research a little easier.

My one concern is that this may be a little non-intuitive/unusual. I don't know of any buttons on any site that store form data in the URL. It always seems to be stored in session (hence my earlier suggestion in #2403 to separate sharing via url from redoing research). From my very little UI design experience, I believe it's better to duplicate behaviors users have learned in other programs. However, since trang thinks that the default behavior should be to not save search criteria and I don't have a strong reason to believe otherwise, this may be the best compromise. So, I think that some explanation would be helpful for users who aren't watching the address bar like a hawk. A popup explaining the button every time you create a template would be annoying, so maybe we could put some alt text on it?

Reply:
Alright. The problem with alt text (I assume you mean a tooltip) is that it won’t show up on mobile. An [md-tooltip](https://material.angularjs.org/latest/api/directive/mdTooltip) will, but only if you tap the thing it is applied to, so it’s no use to apply it to the button. So I suggest putting the tooltip but on a "?" icon next to the button.

I still have one concern. Since there no difference between the pre-filled search form page and the empty search form page (except the URL), the user might get confused. Let’s say I use a templated search at some point, and then I just go on doing other unrelated searches and I forget that the form is pre-filled. Soon, I might get frustrated because the form values keep being pre-filled with unwanted values, even after reloading the page. To avoid this, I think it might be a good idea to have some kind of indication on the pre-filled form that the user is browsing a pre-filled form. It could be as simple as having a "This form is pre-filled. [Clear]" text on the top. Having such text would make the "Your search will remain when you refresh the page or go back to the page" part of the tooltip unnecessary.

About the text, I suggest avoiding the technical word "URL". How about: "Use this button to use the currently select criteria as a base for other searches. You can also bookmark/share the search template with someone else."

Reply:
Okay I like this. I'll rename the button to "Create a search template", add a tooltip, and a little notice that tells you that you're using a template, with a button that will clear the form (just link back to the starting advanced search page).